chia/data_layer/data_layer_wallet.py

```diff
@@ -435,6 +435,7 @@ async def create_update_state_spend(
         )
 
         # Construct the current puzzles
+        assert isinstance(inner_puzzle_derivation.pubkey, G1Element)
```

Comment:
The assertion ought to at least report theoretical ways this could happen, what sort of bug or data corruption might trigger this, etc.  But, shouldn't we know what sort of thing is coming out of `.get_derivation_record_for_puzzle_hash()`?  Why is there uncertainty?

Same goes for all the other cases.